Expensify/App
# Add document picking support to mobile

CC @Jag96 

Add ability to add document attachments to chats on mobile. This already exists for web and desktop, but mobile users were only able to upload images. 
- Uses [DocumentPicker library](https://github.com/rnmods/react-native-document-picker) for native file selection
- Renames our existing ImagePicker class to AttachmentPicker
- Builds custom 'Add document' option into existing lists 


### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/142733

### Tests

**1)** Take Photo
- Press attachment button, then 'Take photo'
- The temporary attachment spinner should display
- Photo should be uploaded as an attachment

**2)** Upload photo from gallery
- Press attachment button, then 'Choose from gallery'
- The temporary attachment spinner should display
- Photo should be uploaded as an attachment

**3)** 
- Press attachment button, then 'Choose Document'
- Select a file for upload (.pdf, .csv, ect)
- File should be uploaded as an attachment

**4)**
- Press attachment button, press cancel button
- The temporary attachment spinner should **not** display

**5)**
- Press attachment button, press 'Choose Document' button
- Press cancel on the modal that pops up
- The temporary attachment spinner should **not** display

**6)**
- Revoke camera permission (or completely reinstall)
- Press attachment button, press 'Take Photo' button
- Deny permission
- The temporary attachment spinner should **not** display


### Screenshots

![Simulator Screen Shot - iPhone 11 - 2020-10-23 at 11 23 02](https://user-images.githubusercontent.com/10736861/96996727-4971c500-1528-11eb-88cc-33bd9514d9fc.png)
![Simulator Screen Shot - iPhone 11 - 2020-10-23 at 12 11 11](https://user-images.githubusercontent.com/10736861/96997038-e3d20880-1528-11eb-9e34-8e013c687e09.png)


**Android**
![Screenshot_1603450582](https://user-images.githubusercontent.com/10736861/96998681-cce0e580-152b-11eb-8514-02fa98d8ee4e.png)
![Screenshot_1603451295](https://user-images.githubusercontent.com/10736861/96996843-7a51fa00-1528-11eb-8f5b-d837a4772bc8.png)
